recipes/diligent-core/all/conanfile.py

```diff
@@ -126,42 +143,22 @@ def _diligent_platform(self):
         elif self.settings.os == "watchOS":
             return "PLATFORM_TVOS"
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["DILIGENT_BUILD_SAMPLES"] = False
-        self._cmake.definitions["DILIGENT_NO_FORMAT_VALIDATION"] = True
-        self._cmake.definitions["DILIGENT_BUILD_TESTS"] = False
-        self._cmake.definitions["DILIGENT_NO_DXC"] = True
-        self._cmake.definitions["DILIGENT_NO_GLSLANG"] = not self.options.with_glslang
-        self._cmake.definitions["SPIRV_CROSS_NAMESPACE_OVERRIDE"] = self.options["spirv-cross"].namespace
-        self._cmake.definitions["BUILD_SHARED_LIBS"] = False
-        self._cmake.definitions["DILIGENT_CLANG_COMPILE_OPTIONS"] = ""
-        self._cmake.definitions["DILIGENT_MSVC_COMPILE_OPTIONS"] = ""
-
-        self._cmake.definitions["ENABLE_RTTI"] = True
-        self._cmake.definitions["ENABLE_EXCEPTIONS"] = True
-
-        self._cmake.definitions[self._diligent_platform()] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
         rename(self, src=os.path.join(self.package_folder, "include", "source_subfolder"),
         dst=os.path.join(self.package_folder, "include", "DiligentCore"))
 
         rmdir(self, os.path.join(self.package_folder, "Licenses"))
         rmdir(self, os.path.join(self.package_folder, "lib"))
         rmdir(self, os.path.join(self.package_folder, "bin"))
-        self.copy("License.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("License.txt", dst="licenses", src=self.source_folder)
```

Comment:
```suggestion
        copy(self, "License.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```
Prefer the copy method. Both self.copy and copy(self are incompatible. The new one does not consider your current context/method, so you need to provide package_folder too.